ovs-dpdk.md

```diff
@@ -0,0 +1,180 @@
+ 
+According to [Intel ONP performance tests](https://download.01.org/packet-processing/ONPS2.1/Intel_ONP_Release_2.1_Performance_Test_Report_Rev1.0.pdf), using OVS with DPDK can provide a huge performance enhancement, increasing network packets throughput and reduced latency.
+ 
+OVS-DPDK has been added to oVirt as an experimental feature.
+Please note that OVS-DPDK cannot access any features located within the Linux kernel, for example: Linux bridge,  tun/tap devices, iptables etc.
+
+## Requirements:
+
+Please follow the instructions in <http://dpdk.org/doc/guides-16.04/linux_gsg/nic_perf_intel_platform.html> in order to acheive best performance.
+ 
+The network card must be on the supported-vendor matrix, located here: <http://dpdk.org/doc/nics>
+
+Please review that your system supports 1GB hugepages:
+```if grep pdpe1gb /proc/cpuinfo >/dev/null 2>&1; then echo "1GB supported."; fi```
+
+* Hardware support: Make sure VT-d / AMD-Vi is enabled in BIOS
+* Kerenel support: When adding a new host via oVirt GUI, go to Hosts -> Edit > Kernel and add the following parameters to Kernel Command line:  ```iommu=pt intel_iommu=on default_hugepagesz=1GB hugepagesz=1G hugepages=N ```
+In case that 1GB hugepages is not supported, please drop ```default_hugepagesz=1GB hugepagesz=1G``` since default hugepages size will be used.
+
+It is best to configure the host before installing it on oVirt-engine, but if changes are made after vdsm is installed, move the host to maintenance mode, reboot the host and reinstall it, for the changes to take effect.
+
+
+## Configure OVS and DPDK on the Host
+
+### Install dpdk packages:
+
+```bash
+# yum install dpdk dpdk-tools driverctl
+
+```
+ 
+### Bind NICs to DPDK:
+
+In order to run a DPDK application, a suitable uio module should be loaded into kernel memory.
+Available modules are: uio_pci_generic, igb_uio and vfio-pci.
+In many cases, the standard uio_pci_generic module included in the Linux kernel can provide the uio capability.
+For some devices which lack support for legacy interrupts, igb_uio should be used, but it’s an out-of-tree kernel module.
+VFIO is preferred to uio drivers in latest DPDK versions in platforms that support VFIO.
+
+All ports that you would like to run DPDK application should be bound to one of the modules before OVS is started. DPDK will ignore all devices which are not bound to one of the drivers above.
+ 
+Load vfio-pci
+
+```
+# modprobe vfio_pci
+```
+Find the appriopriate network devices you wish to bind:
+```
+# lspci | grep Ether
+```
+Review the status of current port bindings:
+```
+# driverctl -v list-devices
+```
+Bind the device to DPDK:
+```
+# driverctl set-override <PCI_ID> vfio-pci
+```
+
+### Hugepages Support
+Hugepages support is required for increasing performance, they allow the Linux kernel to manage larger pages, in addition to the standard 4K pages.
```

Comment:
Suggest:
"Huge pages support is required for increasing performance. It allows the Linux kernel to manage larger pages, in addition to the standard 4K pages."